lib/miq_expression.rb

```diff
@@ -804,17 +804,15 @@ def self.quote(val, typ)
   def self.quote_human(val, typ)
     case typ&.to_sym
     when :integer, :decimal, :fixnum, :float
-      return val.to_i unless val.to_s.number_with_method? || typ == :float
+      return val if val.kind_of?(Numeric)
+      return val.to_i unless val.to_s.number_with_method?
 
-      if val =~ /^([0-9.,]+)\.([a-z]+)$/
-        val, sfx = $1, $2
-        if sfx.ends_with?("bytes") && FORMAT_BYTE_SUFFIXES.key?(sfx.to_sym)
-          "#{val} #{FORMAT_BYTE_SUFFIXES[sfx.to_sym]}"
-        else
-          "#{val} #{sfx.titleize}"
-        end
+      /^([0-9.,]+)\.([a-z]+)$/.match(val.to_s)
```

Comment:
Review this with whitespace removed.  I think it simplifies it.  I was surprised I could not use $1, $2 after line 808 (number_with_method?) since that returns the result of `=~` anyway but alas, they were not set, perhaps when the scope returns out of that method.  It feels redundant to use a nearly identical regex.

btw, any idea why they're different?
https://github.com/ManageIQ/more_core_extensions/blob/27d3e2086d26d7814a1fdeaf9ba7a7281344fab8/lib/more_core_extensions/core_ext/string/to_i_with_method.rb#L3C5-L3C64

number_with_method?
```ruby
/^([0-9\.,]+)\.([a-z]+)$/
```
vs.

this code:
```ruby
/^([0-9.,]+)\.([a-z]+)$/
```

Reply:
```[1] pry(main)> "a" =~ /[.]/
=> nil
[2] pry(main)> "." =~ /[.]/
=> 0
[3] pry(main)> "a" =~ /[\.]/
=> nil
[4] pry(main)> "." =~ /[\.]/
=> 0
```

Reply:
> I was surprised I could not use $1, $2 after line 808 (number_with_method?) since that returns the result of =~ anyway 

Yeah the regex side effects are local scope, if I recall.  Even so, `number_with_method?` should return boolean, and we should provide a separate method that returns the constituent parts (perhaps named `number_with_method` (no `?`).